internal/webhook/feature/features.go

```diff
@@ -47,6 +47,13 @@ const (
 	// This feature gate must be used together with LiteralCertificateSubject webhook feature gate.
 	// See https://github.com/cert-manager/cert-manager/issues/3203 and https://github.com/cert-manager/cert-manager/issues/4424 for context.
 	LiteralCertificateSubject featuregate.Feature = "LiteralCertificateSubject"
+
+	// Owner (responsible for graduating feature through to GA): @inteon
+	// GA: v1.13
+	// DontAllowInsecureCSRUsageDefinition will prevent the webhook from allowing
+	// CertificateRequest's usages to be only defined in the CSR, while leaving
+	// the usages field empty.
+	DontAllowInsecureCSRUsageDefinition featuregate.Feature = "DontAllowInsecureCSRUsageDefinition"
```

Comment:
suggestion: I don't think this flag really matches my understanding of what we want a "feature gate" to be.

My understanding was that a feature gate was something we were concerned might break or need to change in the future.

I don't really see a huge problem here with this just being a regular CLI flag. I suppose a feature gate makes it easier to remove this in the future, but I don't really see a pressing need to remove this option.

It doesn't really matter though!

Reply:
I was thinking that we would like to remove this feature flag in an upcoming release if no one complains that this change causes problems for them. That is why I added this as a feature flag.
I think you only want to be less strict in case you have a cert-manager integration that "incorrectly" uses the API.

Reply:
In the future, I would like to make some other improvements to our validation webhook that might be a bit more difficult to  implement with this feature flag (not sure). eg. https://github.com/cert-manager/cert-manager/pull/6056